repo.go

```diff
@@ -671,6 +672,38 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, length int64, custom json.RawMessage) error {
+	expires := data.DefaultExpires("targets")
+
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+	hashes := make([]string, 1)
+	split_digest := strings.Split(digest, ":")
+	hashes[0] = split_digest[1]
+
+	meta := data.FileMeta{Length: length, Hashes: make(data.Hashes, len(hashes))}
+	meta.Hashes[split_digest[0]], err = hex.DecodeString(hashes[0])
+
+	path := digest
+
+	if len(custom) > 0 {
+		meta.Custom = &custom
+	} else if t, ok := t.Targets[path]; ok {
+		meta.Custom = t.Custom
+	}
+
+	t.Targets[digest] = data.TargetFileMeta{meta}
+
+	t.Expires = expires.Round(time.Second)
+	if _, ok := r.versionUpdated["targets.json"]; !ok {
+		t.Version++
+		r.versionUpdated["targets.json"] = struct{}{}
+	}
+	return r.setMeta("targets.json", t)
```

Comment:
Since this is identical to how `AddTargetsWithExpires` writes targets metadata, can we factor this part out into its own function?